packages/augur-core/source/contracts/trading/ZeroXTrade.sol

```diff
@@ -86,6 +84,89 @@ contract ZeroXTrade is Initializable, IZeroXTrade {
         );
     }
 
+    // ERC1155 Implementation
+    /// @notice Transfers value amount of an _id from the _from address to the _to address specified.
+    /// @dev MUST emit TransferSingle event on success.
+    /// @param from    Source address
+    /// @param to      Target address
+    /// @param id      ID of the token type
+    /// @param value   Transfer amount
+    /// @param data    Additional data with no specified format, sent in call to `_to`
+    function safeTransferFrom(address from, address to, uint256 id, uint256 value, bytes calldata data) external {
+        require(transferFromAllowed);
+        emit TransferSingle(msg.sender, from, to, id, value);
+    }
+
+    /// @notice Send multiple types of Tokens from a 3rd party in one transfer (with safety call).
+    /// @dev MUST emit TransferBatch event on success.
+    /// @param from    Source addresses
+    /// @param to      Target addresses
+    /// @param ids     IDs of each token type
+    /// @param values  Transfer amounts per token type
+    /// @param data    Additional data with no specified format, sent in call to `_to`
+    function safeBatchTransferFrom(address from, address to, uint256[] calldata ids, uint256[] calldata values, bytes calldata data) external {
+        require(transferFromAllowed);
+        emit TransferBatch(msg.sender, from, to, ids, values);
+    }
+
+    /// @notice Get the balance of an account's Tokens.
+    /// @param owner  The address of the token holder
+    /// @param id     ID of the Token
+    /// @return       The _owner's balance of the Token type requested
+    function balanceOf(address owner, uint256 id) external view returns (uint256) {
+        (address _market, uint256 _price, uint8 _outcome, uint8 _type) = unpackTokenId(id);
+        Order.Types _orderType = Order.Types(_type);
+        if (_orderType == Order.Types.Ask) {
+            return askBalance(owner, IMarket(_market), _outcome, _price);
+        } else if (_orderType == Order.Types.Bid) {
+            return bidBalance(owner, IMarket(_market), _outcome, _price);
+        }
+    }
+
+    function bidBalance(address _owner, IMarket _market, uint8 _outcome, uint256 _price) private view returns (uint256) {
+        uint256 _numberOfOutcomes = _market.getNumberOfOutcomes();
+        // Figure out how many almost-complete-sets (just missing `outcome` share) the creator has
+        uint256 _attoSharesOwned = 2**254;
+        for (uint256 _i = 0; _i < _numberOfOutcomes; _i++) {
+            if (_i != _outcome) {
+                uint256 _creatorShareTokenBalance = _market.getShareToken(_i).balanceOf(_owner);
+                _attoSharesOwned = _creatorShareTokenBalance.min(_attoSharesOwned);
+            }
+        }
+
+        uint256 _attoSharesPurchasable = cash.balanceOf(_owner).div(_price);
+
+        return _attoSharesOwned.add(_attoSharesPurchasable);
```

Comment:
Can this overload?

Reply:
Doubtful since youd have to have an impossibly large amount of cash or shares, but if it did it would just make the tx fail as we're using safe math